tests/kubesonic/test_k8s_join_disjoin.py

```diff
@@ -0,0 +1,387 @@
+spec:
+  selector:
+    matchLabels:
+      group: {DAEMONSET_POD_LABEL}
+  template:
+    metadata:
+      labels:
+        group: {DAEMONSET_POD_LABEL}
+    spec:
+      nodeSelector:
+        {DAEMONSET_NODE_LABEL}: "true"
+      hostNetwork: true
+      containers:
+      - image: {DUT_PAUSE_IMAGE}
+        name: {DAEMONSET_CONTAINER_NAME}
+    '''
+
+    vmhost.shell(f"echo -n '{daemonset_content}' > {daemonset_yaml}")
+    vmhost.shell(f"{NO_PROXY} minikube kubectl -- apply -f {daemonset_yaml}")
+    logger.info("Daemonset is deployed")
+
+
+def check_minikube_ready(vmhost):
+    logger.info("Check if minikube is ready")
+    get_minikube_node_cmd = f"{NO_PROXY} minikube kubectl -- get node minikube --no-headers"
+    minikube_status = vmhost.shell(get_minikube_node_cmd, module_ignore_errors=True)
+    status_stdout = minikube_status["stdout"]
+    status_stdout_list = status_stdout.split()
+    if status_stdout != "" and len(status_stdout_list) == 5 and status_stdout_list[1] == "Ready":
+        logger.info("Minikube master is ready")
+        return True
+    else:
+        logger.info("Minikube master is not ready")
+        return False
+
+
+def mark_minikube_started(vmhost):
+    logger.info("Mark minikube as started")
+    vmhost.shell("mkdir -p /run/minikube && touch /run/minikube/started")
+    logger.info("Minikube is marked as started")
+
+
+def mark_minikube_completed(vmhost):
+    logger.info("Mark minikube as completed")
+    vmhost.shell("rm -f /run/minikube/started")
+    logger.info("Minikube is marked as completed")
+
+
+def check_minikube_setup_started(vmhost):
+    logger.info("Check if minikube setup is started")
+    # Check the file createion time
+    minikube_setup_started = vmhost.shell("stat -c %z /run/minikube/started", module_ignore_errors=True)
+    if minikube_setup_started["stdout"] == "":
+        logger.info("Minikube setup is not started")
+        return False, None
+    else:
+        time_format = '%Y-%m-%d %H:%M:%S'
+        creation_time = datetime.strptime(minikube_setup_started["stdout"].strip().split('.')[0], time_format)
+        logger.info(f"Minikube setup is started, creation time is {creation_time}")
+        return True, creation_time
+
+
+def clean_up_and_setup_minikube(vmhost, creds):
+    # Mark the minikube as started
+    mark_minikube_started(vmhost)
+    # Clean up the previous minikube
+    remove_minikube(vmhost)
+    # Download minikube
+    download_minikube(vmhost, creds)
+    # Clean up the setup env
+    remove_k8s_master(vmhost)
+    # Setup k8s master
+    setup_k8s_master(vmhost, creds)
+    # Mark the minikube as completed
+    mark_minikube_completed(vmhost)
+
+
+def clean_configdb_k8s_table(duthost):
+    logger.info("Start to clean k8s table in configdb")
+    duthost.shell("sonic-db-cli CONFIG_DB DEL 'KUBERNETES_MASTER|SERVER'")
+    logger.info("K8s table in configdb is cleaned")
+
+
+@pytest.fixture()
+def setup_and_teardown(duthost, vmhost, creds):
```

Comment:
conftest.py can be used to share fixture across multiple files. This could be helpful as we add more test cases to this directory. This can be done here or in a future PR

ref: https://docs.pytest.org/en/stable/reference/fixtures.html#conftest-py-sharing-fixtures-across-multiple-files
https://github.com/sonic-net/sonic-mgmt/blob/master/tests/generic_config_updater/conftest.py

Reply:
Okay, got it. Very useful info, let's do it in the future if we have more test cases.